### PR TITLE
bpftrace: add page

### PR DESCRIPTION
**Pull request by andrealmeid**
hey there,

bpftrace is a very powerful tool for kernel and low level development, but there are some [many features](https://github.com/iovisor/bpftrace/blob/master/docs/reference_guide.md#usage), that's hard to choose the correct one. looking forward for a feedback!

thanks :)

---
- [X] The page (if new), does not already exist in the repo.
- [X] The page is in the correct platform directory (`common/`, `linux/`, etc.)
- [X] The page has 8 or fewer examples.
- [X] The PR title conforms to the recommended [templates](/tldr-pages/tldr/blob/master/CONTRIBUTING.md#commit-message).
- [X] The page follows the [content guidelines](/tldr-pages/tldr/blob/master/CONTRIBUTING.md#guidelines).
- [X] The page description includes a link to documentation or a homepage (if applicable).

### CODE REVIEW

File: pages/linux/bpftrace.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+`sudo bpftrace -l`
+
+- Run a one-liner program (e.g syscall count by program):
+
+`sudo bpftrace -e 'tracepoint:raw_syscalls:sys_enter { @[comm] = count(); }'`
+
+- Run a program from a file:
+
+`sudo bpftrace {{filename}}`
```
Comment (navarroaxel) [inline pages/linux/bpftrace.md:20]:
```suggestion
`sudo bpftrace {{path/to/file}}`
```

File: pages/linux/bpftrace.md [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+- Trace a program by PID:
+
+`sudo bpftrace -e 'tracepoint:raw_syscalls:sys_enter /pid == {{pid}}/ { @[comm] = count(); }'`
+
+- Do a dry run and prints the output in eBPF format:
```
Comment (navarroaxel) [inline pages/linux/bpftrace.md:26]:
```suggestion
- Do a dry run and display the output in eBPF format:
```

File: pages/linux/bpftrace.md [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+`bpftrace -V`
+
+- List all available probes:
+
+`sudo bpftrace -l`
+
+- Run a one-liner program (e.g syscall count by program):
+
+`sudo bpftrace -e 'tracepoint:raw_syscalls:sys_enter { @[comm] = count(); }'`
```
Comment (navarroaxel) [inline pages/linux/bpftrace.md:16]: The argument should be between `{}`
```suggestion
`sudo bpftrace -e {{'tracepoint:raw_syscalls:sys_enter { @[comm] = count(); }'}}`
```

File: pages/linux/bpftrace.md [hunk 4]
```diff
@@ -0,0 +1,28 @@
+# bpftrace
+
+> High-level tracing language for Linux eBPF.
+> More information: <https://github.com/iovisor/bpftrace>.
+
+- Check bpftrace version:
```
Comment (navarroaxel) [inline pages/linux/bpftrace.md:6]:
```suggestion
- Display bpftrace version:
```

File: pages/linux/bpftrace.md [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+
+`sudo bpftrace {{filename}}`
+
+- Trace a program by PID:
+
+`sudo bpftrace -e 'tracepoint:raw_syscalls:sys_enter /pid == {{pid}}/ { @[comm] = count(); }'`
+
+- Do a dry run and prints the output in eBPF format:
+
+`sudo bpftrace -d -e {{one-line program}}`
```
Comment (navarroaxel) [inline pages/linux/bpftrace.md:28]:
```suggestion
`sudo bpftrace -d -e '{{one_line_program}}'`
```

**Review by navarroaxel — CHANGES_REQUESTED** [review]
Hi @andrealmeid, nice command. I left some comments.

File: pages/linux/bpftrace.md (hunk 5 above)
Comment (andrealmeid) [reply pages/linux/bpftrace.md:28]: Done.

File: pages/linux/bpftrace.md (hunk 4 above)
Comment (andrealmeid) [reply pages/linux/bpftrace.md:6]: Done.

File: pages/linux/bpftrace.md (hunk 3 above)
Comment (andrealmeid) [reply pages/linux/bpftrace.md:16]: Done for both -e arguments

File: pages/linux/bpftrace.md (hunk 2 above)
Comment (andrealmeid) [reply pages/linux/bpftrace.md:26]: Done.

File: pages/linux/bpftrace.md (hunk 1 above)
Comment (andrealmeid) [reply pages/linux/bpftrace.md:20]: Done.

File: pages/linux/bpftrace.md [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+
+`sudo bpftrace -e {{'tracepoint:raw_syscalls:sys_enter { @[comm] = count(); }'}}`
+
+- Run a program from a file:
+
+`sudo bpftrace {{path/to/file}}`
+
+- Trace a program by PID:
+
+`sudo bpftrace -e {{'tracepoint:raw_syscalls:sys_enter /pid == {{pid}}/ { @[comm] = count(); }'}}`
```
Comment (navarroaxel) [inline pages/linux/bpftrace.md:24]: Oh... We should find how to espace the `}` in markdown files for the `{{pid}}` :
Screenshot_20201016_090729

**Review by navarroaxel — CHANGES_REQUESTED** [review]
Thanks @andrealmeid, but we have an issue with the markdown

File: pages/linux/bpftrace.md (hunk 6 above)
Comment (andrealmeid) [reply pages/linux/bpftrace.md:24]: @navarroaxel so there's no support for nested `{{ }}`? maybe we should just replace `{{pid}}` for something like `PID`

Comment (mebeim) [reply pages/linux/bpftrace.md:24]: We cannot do much about escaping. Each client may parse commands in a slightly different way, and nested braces support is up to the particular implementation. There is no standard escape syntax. 

In order to get around this, I would suggest one of the following:

1. Remove the outermost `{{` and `}}` braces, and only keep them around `pid`.
2. Remove the braces around `pid` and only keep the outermost ones. Change `pid` to `PID_TO_TRACE`.

**Comment by andrealmeid** [comment]
@navarroaxel: I applied the 2nd suggestion from @mebeim (thanks!)

File: pages/linux/bpftrace.md [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+
+`sudo bpftrace {{path/to/file}}`
+
+- Trace a program by PID:
+
+`sudo bpftrace -e {{'tracepoint:raw_syscalls:sys_enter /pid == 123/ { @[comm] = count(); }'}}`
+
+- Do a dry run and display the output in eBPF format:
+
+`sudo bpftrace -d -e {{one_line_program}}`
```
Comment (navarroaxel) [inline pages/linux/bpftrace.md:28]: Just add the quotes to the argument because need it.
```suggestion
`sudo bpftrace -d -e '{{one_line_program}}'`
```

**Review by navarroaxel — CHANGES_REQUESTED** [review]

File: pages/linux/bpftrace.md (hunk 7 above)
Comment (andrealmeid) [reply pages/linux/bpftrace.md]: I added inside of `{{' ... '}}`, just like the other examples on this file

Comment (navarroaxel) [reply pages/linux/bpftrace.md]: We had a discuss this in #4363 and the quotes should be outside `{{}}`. Could you please update the 3 example descriptions?

Comment (navarroaxel) [reply pages/linux/bpftrace.md]: @andrealmeid We had a discuss about this in #4363 and the quotes should be outside `'{{...}}'`. Could you please update the 3 example descriptions? And please *DO NOT* push force, is harder to us to track changes.

Comment (andrealmeid) [reply pages/linux/bpftrace.md]: @navarroaxel ok, sorry, I won't push force again

**Review by navarroaxel — APPROVED** [review]
LGTM, thanks @andrealmeid :grin:

**Review by mebeim — APPROVED** [review]
Thank you @andrealmeid :)

**Pull request merged**